app/uk/gov/hmrc/entrydeclarationstore/services/EntryDeclarationStore.scala

```diff
@@ -63,69 +64,85 @@ class EntryDeclarationStoreImpl @Inject()(
     implicit hc: HeaderCarrier): Future[Either[ErrorWrapper[_], SuccessResponse]] =
     timeFuture("Service handleSubmission", "handleSubmission.total") {
 
-      val correlationId          = idGenerator.generateCorrelationId
-      val submissionId           = idGenerator.generateSubmissionId
-      val receivedDateTime       = Instant.now(clock)
-      val input: InputParameters = InputParameters(mrn.isDefined, submissionId, correlationId, receivedDateTime)
+      val correlationId    = idGenerator.generateCorrelationId
+      val submissionId     = idGenerator.generateSubmissionId
+      val receivedDateTime = Instant.now(clock)
+
+      implicit val lc: LoggingContext =
+        LoggingContext(correlationId = Some(correlationId), submissionId = Some(submissionId))
+
+      ContextLogger.info("Handling submission")
```

Comment:
Given this logging context isn't used anywhere else, is it clearer to pass the parameter in here. You create an implicit loggingContext later on which makes sense.
```suggestion
      ContextLogger.info("Handling submission")(LoggingContext(correlationId = Some(correlationId), submissionId = Some(submissionId)))
```